core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraFromRjsAdapter.kt

```diff
@@ -553,57 +606,16 @@ fun adaptRawInfra(infra: SignalingInfra, rjsInfra: RJSInfra): SimInfraAdapter {
             }
         }
     }
-
-    // parse switches
     for (switchEntry in infra.switches) {
         val oldSwitch = switchEntry.value!!
-        val nodeGraph = oldSwitch.graph!!
-        val infraGraph = infra.trackGraph!!
-        trackNodeMap[oldSwitch] =
-            builder.movableElement(oldSwitch.id, oldSwitch.groupChangeDelay.seconds) {
-                val nodeMap: MutableMap<SwitchPort, TrackNodePortId> = mutableMapOf()
-                for (node in nodeGraph.nodes()) {
-                    var track: TrackEdge? = null
-                    for (edge in infraGraph.incidentEdges(node)) {
-                        if (edge is UndirectedTrackSection) {
-                            track = edge
-                            break
-                        }
-                    }
-                    track!!
-                    val endpoint =
-                        if (infraGraph.incidentNodes(track).nodeU() == node) Endpoint.START
-                        else Endpoint.END
-                    nodeMap[node] =
-                        port(EndpointTrackSectionId(oldTrackSectionMap[track]!!, endpoint))
-                }
-                val switchGroups = mutableMapOf<String, TrackNodeConfigId>()
-                for (group in oldSwitch.groups.entries()) {
-                    val groupName = group.key!!
-                    val nodes = nodeGraph.incidentNodes(group.value!!)
-                    val portPair = Pair(nodeMap[nodes.nodeU()]!!, nodeMap[nodes.nodeV()]!!)
-                    switchGroups[groupName] = config(groupName, portPair)
-                }
-                trackNodeGroupsMap[oldSwitch] = switchGroups
-            }
-    }
+        trackNodeMap[oldSwitch] = nodeNameToIdxMap[oldSwitch.id]
 
-    // parse track links
-    for (node in infra.trackGraph.nodes()) {
-        if (node is Joint) {
-            val edges = infra.trackGraph.incidentEdges(node)
-            assert(edges.count() == 2)
-            builder.movableElement(node.id, Duration.ZERO) {
-                val ports = ArrayList<TrackNodePortId>()
-                for (edge in edges) {
-                    val endpoint =
-                        if (infra.trackGraph.incidentNodes(edge).nodeU() == node) Endpoint.START
-                        else Endpoint.END
-                    ports.add(port(EndpointTrackSectionId(oldTrackSectionMap[edge]!!, endpoint)))
-                }
-                config("default", Pair(ports[0], ports[1]))
-            }
+        val switchGroups = mutableMapOf<String, TrackNodeConfigId>()
+        val configs = builder.getTrackNodePool().get(nodeNameToIdxMap[oldSwitch.id]!!).configs
```

Comment:
```suggestion
        val configs = builder.getTrackNodePool()[nodeNameToIdxMap[oldSwitch.id]!!].configs
```

Reply:
Done in https://github.com/osrd-project/osrd/pull/6799/commits/c4d0a6a9f7a45fa3697f8a392413181cd6cc6e5d